src/app/profile/yourGiving/givingRecipientView/givingRecipientView.component.js

```diff
@@ -0,0 +1,49 @@
+import angular from 'angular';
+import donationRecipient from './donationRecipient/donationRecipient.component';
+import donationsService from 'common/services/api/donations.service';
+import template from './givingRecipientView.tpl';
+
+let componentName = 'givingRecipientView';
+
+class GivingRecipientView {
+
+  /* @ngInject */
+  constructor( donationsService ) {
+    this.donationsService = donationsService;
+  }
+
+  $onChanges( changes ) {
+    if ( angular.isDefined( changes.filter ) ) {
+      this.loadRecipients( changes.filter.currentValue == 'recent' ? undefined : changes.filter.currentValue );
+    }
+  }
+
+  loadRecipients( year ) {
+    this.setLoading( {loading: true} );
+    this.recipients = [];
+    if ( angular.isDefined( this.subscriber ) ) this.subscriber.unsubscribe();
+    this.subscriber = this.donationsService.getRecipients( year ).subscribe( ( recipients ) => {
```

Comment:
Canceling the current subscriber if filter changes happen while it's waiting on a response. I know this doesn't cancel the http request, but it does ignore the response when it come back. This prevents a race condition where we could end up showing results for the previous filter and no the current one.